System/Clipboard/X11.hs

```diff
@@ -0,0 +1,93 @@
+module System.Clipboard.X11
+( getClipboardString
+, setClipboardString
+) where
+
+import Graphics.X11.Xlib
+import Graphics.X11.Xlib.Extras
+import System.Posix.Process (forkProcess)
+
+import Data.Maybe
+import Data.Functor
+import Control.Monad
+import Foreign.C.Types (CChar, CUChar)
+import Foreign.Marshal.Array (withArrayLen)
+import Codec.Binary.UTF8.String (decode, encode)
+
+getClipboardString :: IO (Maybe String)
+getClipboardString = withInitialSetup $ \display window clipboard -> do
+    inp <- internAtom display "clipboard_get" False
+    target <- internAtom display "UTF8_STRING" True
+    xConvertSelection display clipboard target inp window currentTime
+    clipboardInputWait display window inp
+
+clipboardInputWait :: Display -> Window -> Atom -> IO (Maybe String)
+clipboardInputWait display window inp = do
+    ev <- getNextEvent display
+    if ev_event_type ev == selectionNotify
+        then (charsToString <$>) <$> getWindowProperty8 display inp window
+        else clipboardInputWait display window inp
+
+charsToString :: [CChar] -> String
+charsToString = decode . map fromIntegral
+
+
+setClipboardString :: String -> IO ()
+setClipboardString str = void $ forkProcess $
```

Comment:
Why do we need ``forkProcess`` here? This will stop exceptions to propagate, so the user of the library won't be able to catch them, or even being able to tell if the operation was successful. Or am I missing something?